src/main/java/Duke.java

```diff
@@ -11,46 +11,68 @@ public static void main(String[] args) {
-                String[] words = input.split(" ");
-                if(words[0].equals("done")) {
-                    // Check if invalid done command is given
-                    if(words.length < 2) {
-                        System.out.println("Invalid command!");
-                    } else {
-                        int doneTask = Integer.parseInt(words[1]) - 1;
-                        list[doneTask].markAsDone();
-                        System.out.println("Nice! I've marked this task as done:");
-                        list[doneTask].printTask();
+                // Check for other commands
+                switch(input) {
+                case ("list"):
+                    System.out.println("Here are the tasks in your list:");
+                    for(int i = 0;i < inputs; i++) {
+                        System.out.print((i+1) + ".");
+                        System.out.println(list[i].toString());
                     }
-                } else {
-                    // Echo input by user
-                    System.out.println(input);
-                    Task t = new Task(input);
-                    list[inputs] = t;
+                    break;
+
+                case ("done"):
+                    int done = sc.nextInt();
+                    list[done - 1].markAsDone();
+                    System.out.println("Nice! I've marked this task as done:");
+                    System.out.println(list[done - 1].toString());
+                    break;
+
+                case ("todo"):
+                    String description = sc.nextLine();
+                    list[inputs] = new ToDo(description);
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println(list[inputs].toString());
+                    System.out.println("Now you have " + (inputs + 1) + " tasks in the list.");
                     inputs++;
+                    break;
+
+                case ("deadline"):
+                    String desc = sc.nextLine();
+                    String[] splitDesc = desc.split("/by");
+                    list[inputs] = new Deadline(splitDesc[0], splitDesc[1]);
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println(list[inputs].toString());
+                    System.out.println("Now you have " + (inputs + 1) + " tasks in the list.");
+                    inputs++;
+                    break;
+
+                case ("event"):
+                    String event = sc.nextLine();
+                    String[] splitEvent = event.split("/at");
+                    list[inputs] = new Event(splitEvent[0],splitEvent[1]);
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println(list[inputs].toString());
+                    System.out.println("Now you have " + (inputs + 1) + " tasks in the list.");
+                    inputs++;
+                    break;
```

Comment:
For todo, deadline and event cases in switch, there is duplication of the following code
                    System.out.println("Got it. I've added this task:");
                    System.out.println(list[inputs].toString());
                    System.out.println("Now you have " + (inputs + 1) + " tasks in the list.");
                    inputs++;
Try to consolidate this repeated code into a single function eg. void printAddedTask(Task task); for all the System.out functions then call this function for each case. This will help minimise duplication of the code and make your main function more readable.